pkg/framework/deployment/deployment.go

```diff
@@ -0,0 +1,69 @@
+package deployment
+
+import (
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Config struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type Deployment struct {
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Deployment, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	d := &Deployment{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return d, nil
+}
+
+// Check ensures that key properties of the deployment are correct.
+func (d *Deployment) Check(name string, replicas int, expectedLabels, expectedMatchLabels map[string]string) error {
+	ds, err := d.k8sClient.Apps().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Maskf(notFoundError, "could not find deployment: '%s'", name)
```

Comment:
This error is printed like: `not found: could not find deployment: '...'`. Let's not repeat this information.`not found: deployment: '...'` is enough.

Reply:
Good catch, done.